src/lem/coroutine/synthesis.rs

```diff
@@ -484,19 +484,8 @@ fn synthesize_run<'a, F: LurkField, CS: ConstraintSystem<F>>(
             Op::Open(sec, pay, comm) => {
                 let comm = bound_allocations.get_ptr(comm)?;
                 let comm_tag = g.alloc_tag(&mut cs, &Comm);
-                let preimg_alloc_nums = if let Some(comm_val) = comm.get_value::<Tag>() {
-                    let raw_ptr = store.to_raw_ptr(&FWrap(*comm_val.value()));
-                    let idx = raw_ptr.get_hash4().unwrap();
-                    store
-                        .expect_raw_ptrs::<4>(idx)
-                        .iter()
-                        .enumerate()
-                        .map(|(i, raw)| {
-                            let cs = ns!(cs, format!("preimg {i}"));
-                            let z = store.hash_raw_ptr(raw).0;
-                            AllocatedNum::alloc_infallible(cs, || z)
-                        })
-                        .collect::<Vec<_>>()
+                let preimg_alloc_nums = if let Some(_comm_val) = comm.get_value() {
+                    todo!()
```

Comment:
Was this meant to be left as `todo!()`?

Reply:
The original code was just wrong. I'm still deciding what to do about it. Maybe I should just leave it as a todo and open an issue